File: apps/web/lib/event-labels.ts
export function eventPhaseLabel(phase: unknown): string {
  switch (phase) {
    case 'UPCOMING': return 'Upcoming';
    case 'CURRENT': return 'Happening now';
    case 'COMPLETED': return 'Past event';
    case 'CANCELLED': return 'Cancelled';
    case 'POSTPONED': return 'Postponed';
    case 'UNSCHEDULED': return 'Date to be announced';
    default: return 'Date not confirmed';
  }
}

export function eventRegistrationLabel(phase: unknown, status: unknown): string | undefined {
  if (phase !== 'UPCOMING' && phase !== 'CURRENT') return undefined;
  if (status === 'OPEN') return 'Registration listed open';
  if (status === 'CLOSED') return 'Registration closed';
  return undefined;
}

File: apps/fitcalgary_app/lib/features/events/event_labels.dart
String eventPhaseLabel(String? phase) => switch (phase) {
  'UPCOMING' => 'UPCOMING',
  'CURRENT' => 'HAPPENING NOW',
  'COMPLETED' => 'PAST EVENT',
  'CANCELLED' => 'CANCELLED',
  'POSTPONED' => 'POSTPONED',
  'UNSCHEDULED' => 'DATE TO BE ANNOUNCED',
  _ => 'DATE NOT CONFIRMED',
};

String? eventRegistrationLabel(String? phase, String? status) {
  if (phase != 'UPCOMING' && phase != 'CURRENT') return null;
  return switch (status) {
    'OPEN' => 'REGISTRATION LISTED OPEN',
    'CLOSED' => 'REGISTRATION CLOSED',
    _ => null,
  };
}
